### PR TITLE
Janet playground merged into Janetdocs WARNING -- big changes here

### PR DESCRIPTION
**Pull request by MikeBeller**
Hi Sean.  Here is a PR to merge the Janet Playground into Janetdocs.  Obviously there are big changes here, and you should not feel like I'm expecting you to accept them if it takes Janetdocs in a direction you don't like.  You can also just clone my fork of janetdocs, and checkout it's playground branch, and test that and see if you like it before deciding.

Here is what's in there:

- Added a "Playground" link to each example -- pops open the Playground with that specific example code in it
- Added a "Playground" link to the main menu -- allows you to go directly to the playground without picking an example
- Added ACE code editor to playground, with clojure mode and 2-space indents (does paren matching too)
- Added a "Format" button to the playground which autoformats your code (with spork/fmt/format)
- Fixed a broken 404 handler in janetdocs (you were returning status 200 on 404 errors even though the HTML said otherwise)

NOTES:
- I simply could not get the Playground to work with turbolinks -- so I disabled it everywhere. Tried some disabling tags from the turbolinks docs but nope.  If you can figure it out I'm happy to revert.  Wasted much time on this though.
- Halo2 doesn't support wasm mime type.  I posted an issue in halo2.  It's not fatal, but inefficient because the playground ends up retrieving the wasm file twice.
- Right now to update the Janet version the playground uses requires checking out my janet-playground project, jpm building it, and copying out the wasm/js files in build/ that it generates.  If you do end up merging my PR, then I can streamline the process for you but I'd probably want to hear from you what would work for you.
- The usual caveats for WASM janet apply -- not all functions work (specifically all the ev/ stuff for example)

POSSIBLE NEXT STEPS:

- Add a "post" handler so other sites can open the playground with their own code in the post input
- Add a "SAVE" button which creates a permalink that people can share.  This would be for non-persistent examples.  E.g. I am on gitter and want to share a code example with someone.  I put it in the playground, press SAVE, and share the permalink with the other party.  Could create a lot of entries in the DB but you could purge them after some reasonable amount of time.

Cheers

Mike

### CODE REVIEW

**Comment by pepe** [comment]
This is fine work.

BTW, I did PR to add wasm to mime types of halo2. Feel free to review.

File: routes/playground.janet [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        [:spacer]]
+      [:div {:id "code" :class "hljs" :style "height:60vh;"} (raw code)]
+      [:pre {:id "output" :style "overflow:auto;"}]
+      [:div {:id "hiddencode" :style "display:none;"}]
+      [:div {:id "sporkformat" :style "display:none;"}
+       (string (slurp "public/playground/fmt.janet"))]
+    [:script {:type "text/javascript" :src "/playground/playground.js" :async "false"}]
+    [:script {:type "text/javascript" :src "/playground/janet.js" :async "false"}]
+    [:script {:type "text/javascript" :src "/playground/jdocs_playground.js" :async "false"}] 
+    ])
```
Comment (swlkr) [inline routes/playground.janet:21]: It's not a big deal but you could use something like [parinfer](https://shaunlebron.github.io/parinfer/) to align these parens

**Comment by swlkr** [comment]
I finally managed to get this cloned and checked it out, this is slick!

Seriously, this is incredible. There are some loose ends, but I can tie those up.

:shipit: 🚀 🚢

**Comment by MikeBeller** [comment]
Great -- glad you like it.  Once it's up on janetdocs and people have a
chance to try it out, I will put in some work on adding the "ad hoc
sharable examples", and also figure out a way to make it easier for you to
pull in a new version of janet.wasm when you want to upgrade the version.

**Pull request merged**